test/e2e/storage/volumes.go

```diff
@@ -518,7 +518,11 @@ var _ = framework.KubeDescribe("Volumes [Volume]", func() {
 			volumeName, err := framework.CreatePDWithRetry()
 			Expect(err).NotTo(HaveOccurred())
 			defer func() {
-				framework.ExpectNoError(framework.DeletePDWithRetry(volumeName))
+				// - Get NodeName from the pod spec to which the volume is mounted.
+				// - Force detach and delete.
+				pod, err := f.PodClient().Get(config.Prefix+"-client", metav1.GetOptions{})
+				Expect(err).NotTo(HaveOccurred(), "Failed getting pod %q.", config.Prefix+"-client")
```

Comment:
Question: if the pod Get() fails is there a way to detatch / delete the PD without having a current copy of the pod? Eg. if the pod's name and ns are known, is there a func that detatches/deletes based on those fields?